### PR TITLE
Run visual regression tests to find regression bugs of visualization module.

### PR DESCRIPTION
**Pull request by c-bata**
I once created #3711 to introduce visual regression tests. This PR added a CircleCI's workflow config only.

## Motivation

It is difficult to ensure the behavior of the visualization module with unit tests, which can easily lead to test case omissions and increased complexity. To solve this problem, I created https://github.com/c-bata/optuna-visualization-regression-tests. This tool prepares various studies and executes each function in Optuna's visualization module. The generated images can be viewed in HTML as shown below.

plot_slice

See https://optuna.github.io/optuna-visualization-regression-tests/ for details.

This PR makes it easier for reviewers to notice regression bugs by running this tool on every new PR created in Optuna and uploading the results to the CircleCI Artifact.

FYI: This tool actually helped me find the following bug.
https://github.com/optuna/optuna/issues/3681

## Description of the changes

Add CircleCI workflow to execute https://github.com/optuna/optuna-visualization-regression-tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,3 +45,43 @@ jobs:
...
+          name: build
+          command: |
+            . venv/bin/activate
+            cd optuna-visualization-regression-tests
+            python visual_regression_tests.py --heavy --output-dir ./public
+          environment:
+            OMP_NUM_THREADS: 1
+
+      - store_artifacts:
+          path: ./optuna-visualization-regression-tests/docs
```
Comment (c-bata) [inline .circleci/config.yml:87]:
```suggestion
          path: ./optuna-visualization-regression-tests/public
```

**Comment by c-bata** [comment]
I confirmed that the workflow works as expected.

* https://app.circleci.com/pipelines/github/optuna/optuna/9020/workflows/02facea8-f95e-4862-9fce-f712ed59b8c1/jobs/70585
* https://output.circle-artifacts.com/output/job/10959dfc-f04c-494d-930c-d4c3eaa42beb/artifacts/0/optuna-visualization-regression-tests/public/index.html

Please note that `.github/workflows/circleci-artifacts-link.yml` isn't triggered until merged into master branch since the trigger condition is `on: status`.

> Note: This event will only trigger a workflow run if the workflow file is on the default branch.
> https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#status

**Review by HideakiImamura — APPROVED** [review]
Thanks for the PR! LGTM!

As a followup, it would be great to add a documentation to use the visual regression test for reviewers. (Maybe, is is on the wiki?)

**Review by contramundum53 — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**